checker/check_result.go

```diff
@@ -63,6 +63,32 @@ const (
 	FileTypeURL
 )
 
+// OutcomeType represents the outcome of a result.
+type OutcomeType int
+
+const (
+	// OutcomeTypeUnknown is for unknown type. Used for debugging and catching unset types.
+	OutcomeTypeUnknown OutcomeType = iota
```

Comment:
Nit: type is already `OutcomeType`, so rename the enums to `Unknown`, `Debug` etc. to avoid redundancy.

Also, note that when this gets printed out it'll only print out an integer. Consider using a string type to make this clearer.

Reply:
I tend to use `OutcomeType` as a namespace, so it does not collide with other names. I can change that, no problem.
We don't print this value yet, the plan is to have a `outcomeToString()` function for it.

Reply:
removed it from this PR.